src/main/java/org/elasticsearch/index/search/geo/GeoPolygonFilter.java

```diff
@@ -106,4 +136,41 @@ private static boolean pointInPolygon(GeoPoint[] points, double lat, double lon)
             return inPoly;
         }
     }
+
+    private BoundingBox getBoundingBox(GeoPoint[] points) {
+        Validate.isTrue(points.length > 0);
+        double maxLat = points[0].lat();
+        double minLat = points[0].lat();
+        double maxLon = points[0].lon();
+        double minLon = points[0].lon();
+
+        for (int i = 1; i < points.length; i++) {
+            maxLat = Math.max(maxLat, points[i].lat());
+            minLat = Math.min(minLat, points[i].lat());
+            maxLon = Math.max(maxLon, points[i].lon());
+            minLon = Math.min(minLon, points[i].lon());
+        }
+
+        GeoPoint topLeft = new GeoPoint(maxLat, minLon);
+        GeoPoint bottomRight = new GeoPoint(minLat, maxLon);
+        return new BoundingBox(topLeft, bottomRight);
+    }
+
+    private static class BoundingBox {
```

Comment:
Did you want me to take a stab at eliminating the duplication, or should I leave it for now since @nknize is planning a refactor?